astropy/io/ascii/cds.py

```diff
@@ -164,14 +210,437 @@ def get_cols(self, lines):
+
+            # Find format of the Float string
+            fmt = self._split_float_format(str(rec))
+
+            # If value is in Scientific notation
+            if fmt[4] is True:
+                # if the previous column value was in normal Float format
+                # set maxsize, maxprec and maxdec to default.
+                if fformat == 'F':
+                    maxsize, maxprec, maxdec = 1, 0, 0
+                # Designate the column to be in Scientific notation.
+                fformat = 'E'
+            else:
+                # Move to next column value if
+                # current value is not in Scientific notation
+                # but the column is designated as such because
+                # one of the previous values was.
+                if fformat == 'E':
+                    continue
+
+            if maxsize < fmt[0]:
+                maxsize = fmt[0]
+            if maxent < fmt[1]:
+                maxent = fmt[1]
+            if maxdec < fmt[2]:
+                maxdec = fmt[2]
+            if fmt[3]:
+                sign = True
+
+            if maxprec < fmt[1] + fmt[2]:
+                maxprec = fmt[1] + fmt[2]
+
+        if fformat == 'E':
+            col.meta.size = maxsize
+            if sign:
+                col.meta.size += 1
+            # Number of digits after decimal is replaced by the precision
+            # for values in Scientific notation, when writing that Format.
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxprec)
+            col.format = str(col.meta.size) + "." + str(maxdec) + "e"
+        else:
+            col.meta.size = maxent + maxdec + 1
+            if sign:
+                col.meta.size += 1
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxdec)
+            col.format = col.fortran_format[1:] + "f"
 
-class CdsData(core.BaseData):
+    def write_byte_by_byte(self):
+        """
+        Writes the Byte-By-Byte description of the table.
+
+        Columns that are `astropy.coordinates.SkyCoord` or `astropy.time.TimeSeries`
+        objects or columns with values that are such objects are recognized as such,
+        and some predefined labels and description is used for them.
+        See the Vizier CDS Standard documentation in the link below for more details
+        on these. An example Byte-By-Byte table is shown here.
+
+        See: http://vizier.u-strasbg.fr/doc/catstd-3.1.htx
+
+        Example::
+
+        --------------------------------------------------------------------------------
+        Byte-by-byte Description of file: table.dat
+        --------------------------------------------------------------------------------
+        Bytes Format Units  Label     Explanations
+        --------------------------------------------------------------------------------
+         1- 8   A8     ---    names    Description of names
+        10-14   E5.1   ---    e       [-3160000.0/0.01] Description of e
```

Comment:
_pycdsreadme_ because of its string formatting does this. It is not required by the format (http://cdsarc.u-strasbg.fr/ftp/cats/I/221/ReadMe). This docstring example is outdated.